code/modules/ghosttrap/trap.dm

```diff
@@ -21,7 +21,7 @@ var/list/ghost_traps
 
 /datum/ghosttrap
 	var/object = "positronic brain"
-	var/minutes_since_death = 0     // If non-zero the ghost must have been dead for this many minutes to be allowed to spawn
```

Comment:
this minutes since death feature wasn't being used and seems dumb.

It you want custom checks, a new type can be defined